pkg/assets/apiserver/asset_apiserver.go

```diff
@@ -55,11 +58,19 @@ func init() {
 
 const (
 	OpenShiftWebConsoleClientID = "openshift-web-console"
+
+	// Discovery endpoint for OAuth 2.0 Authorization Server Metadata
+	// See IETF Draft:
+	// https://tools.ietf.org/html/draft-ietf-oauth-discovery-04#section-2
+	oauthMetadataEndpoint = "/.well-known/oauth-authorization-server"
 )
 
 type ExtraConfig struct {
 	Options   v1.WebConsoleConfiguration
 	PublicURL url.URL
+
+	OAuthAuthorizationEndpoint string
+	OAuthTokenEndpoint         string
```

Comment:
I expected to see a polling loop in this pull that watches to see when the value changes and either live updates or suicides.  The value can change over time, same as anything else that can rotate.